File: core-github/src/main/java/me/ghostbear/kumaslash/github/model/PullRequest.java
package me.ghostbear.kumaslash.github.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.jetbrains.annotations.Nullable;

import java.util.Objects;

@JsonIgnoreProperties(ignoreUnknown = true)
public record PullRequest(
		@Nullable
		@JsonProperty("html_url")
		String htmlUrl,
		@Nullable
		@JsonProperty("merged_at")
		String mergedAt
) {

	public boolean isMerged() {
		return Objects.nonNull(mergedAt);
	}

}

File: data/src/main/kotlin/me/ghostbear/data/github/model/PullRequest.kt
package me.ghostbear.data.github.model


import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PullRequest(
    @SerialName("url")
    val url: String = "",
    @SerialName("html_url")
    val htmlUrl: String = "",
    @SerialName("diff_url")
    val diffUrl: String = "",
    @SerialName("patch_url")
    val patchUrl: String = "",
    @SerialName("merged_at")
    val mergedAt: String? = null
)